controllers/issuer_controller.go

```diff
@@ -69,51 +76,58 @@ type IssuerReconciler struct {
 	PostSetupWithManager func(context.Context, schema.GroupVersionKind, ctrl.Manager, controller.Controller) error
 }
 
-func (r *IssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, returnedError error) {
+func (r *IssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx).WithName("Reconcile")
 
 	logger.V(2).Info("Starting reconcile loop", "name", req.Name, "namespace", req.Namespace)
 
-	result, issuerStatusPatch, returnedError := r.reconcileStatusPatch(logger, ctx, req)
-	logger.V(2).Info("Got StatusPatch result", "result", result, "patch", issuerStatusPatch, "error", returnedError)
-	if issuerStatusPatch != nil {
-		cr, patch, err := ssaclient.GenerateIssuerStatusPatch(r.ForObject, req.Name, req.Namespace, issuerStatusPatch)
-		if err != nil {
-			return ctrl.Result{}, utilerrors.NewAggregate([]error{err, returnedError})
-		}
+	result, issuerStatusPatch, reconcileError := r.reconcileStatusPatch(logger, ctx, req)
+
+	if issuerStatusPatch == nil {
+		logger.V(2).Info("Got nil StatusPatch result", "result", result, "error", reconcileError)
+		return ctrl.Result{}, reconcileError
+	}
+
+	obj, patch, err := ssaclient.GenerateIssuerStatusPatch(r.ForObject, req.Name, req.Namespace, issuerStatusPatch)
+	if err != nil {
+		return ctrl.Result{}, utilerrors.NewAggregate([]error{err, reconcileError}) // requeue with backoff
+	}
 
-		if err := r.Client.Status().Patch(ctx, cr, patch, &client.SubResourcePatchOptions{
-			PatchOptions: client.PatchOptions{
-				FieldManager: r.FieldOwner,
-				Force:        ptr.To(true),
-			},
-		}); err != nil {
-			if err := client.IgnoreNotFound(err); err != nil {
-				return ctrl.Result{}, utilerrors.NewAggregate([]error{err, returnedError})
-			}
-			logger.V(1).Info("Not found. Ignoring.")
+	logger.V(2).Info("Got StatusPatch result", "result", result, "error", reconcileError, "patch", patch)
+
+	if err := r.Client.Status().Patch(ctx, obj, patch, &client.SubResourcePatchOptions{
+		PatchOptions: client.PatchOptions{
+			FieldManager: r.FieldOwner,
+			Force:        ptr.To(true),
+		},
+	}); err != nil {
+		if err := client.IgnoreNotFound(err); err != nil {
```

Comment:
This `IgnoreNotFound` func is confusing, and assigning `err` and doing `if err != nil` on the same line makes it also harder to understand what is going on. 

What about returning a boolean to make it clearer what we are doing here?

```go
	err := r.Client.Status().Patch(ctx, obj, patch, &client.SubResourcePatchOptions{
		PatchOptions: client.PatchOptions{
			FieldManager: r.FieldOwner,
			Force:        ptr.To(true),
		},
	})
	if err != nil {
	    if client.IsNotFound(err) {
			return ctrl.Result{}, utilerrors.NewAggregate([]error{err, reconcileError}) // requeue with backoff
		}

		logger.V(1).Info("Issuer not found. Ignoring.")
		return ctrl.Result{}, nil // done
	}
```